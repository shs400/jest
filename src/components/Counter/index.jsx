import React, { Component } from 'react';
import { Button } from '../Shared';
import PropTypes from "prop-types";

class Counter extends Component {
  static propTypes = {
    counter: PropTypes.object,
  }

  static defaultProps = {
    counter: {},
  }
  constructor(props) {
    super(props);
    this.state = {
      value: 0,
    };
  }

  onIncrease = () => {
    this.props.counter.increment();
    this.setState(({ value }) => ({ value: value + 1 }));
  };

  onDecrease = () => {
    this.props.counter.decrement();
    this.setState(({ value }) => ({ value: value <= 0 ? value : value - 1 }));
  };

  render() {
    const { value } = this.state;
    return (
      <div>
        <h1>카운터</h1>
        <h2>{value}</h2>
        <div>
          <Button
            type="button"
            className="increase"
            onClick={this.onIncrease}
            label="+"
          />
          <Button
            type="button"
            className="decrease"
            onClick={this.onDecrease}
            label="-"
          />
        </div>
      </div>
    );
  }
}

export default Counter;
