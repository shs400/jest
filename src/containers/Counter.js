import React  from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import Counter from '../components/Counter';
import { increment, decrement } from '../actions';

const mapStateToProps = state => ({
  value: state.counter.value
});

const mapDispatchToProps = dispatch => ({
  counter: bindActionCreators({ increment, decrement }, dispatch)
});

export default connect(mapStateToProps, mapDispatchToProps)(Counter);