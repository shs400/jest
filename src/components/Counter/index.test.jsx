import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Counter from './index';
configure({ adapter: new Adapter() });

describe('Counter', () => {
  let component = null;

  it('renders correctly', () => {
    component = shallow(<Counter />);
  });

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('increases correctly', () => {
    component.instance().onIncrease();
    expect(component.state().value).toBe(1);
    expect(component).toMatchSnapshot();
  });

  it('decreases correctly', () => {
    component.instance().onDecrease();
    expect(component.state().value).toBe(0);
    expect(component).toMatchSnapshot();
  });
});