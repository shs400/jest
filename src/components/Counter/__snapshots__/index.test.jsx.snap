// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter decreases correctly 1`] = `
<div>
  <h1>
    카운터
  </h1>
  <h2>
    1
  </h2>
  <div>
    <Button
      className="increase"
      label="+"
      onClick={[Function]}
      type="button"
    />
    <Button
      className="decrease"
      label="-"
      onClick={[Function]}
      type="button"
    />
  </div>
</div>
`;

exports[`Counter increases correctly 1`] = `
<div>
  <h1>
    카운터
  </h1>
  <h2>
    2
  </h2>
  <div>
    <Button
      className="increase"
      label="+"
      onClick={[Function]}
      type="button"
    />
    <Button
      className="decrease"
      label="-"
      onClick={[Function]}
      type="button"
    />
  </div>
</div>
`;

exports[`Counter matches snapshot 1`] = `
<div>
  <h1>
    카운터
  </h1>
  <h2>
    0
  </h2>
  <div>
    <Button
      className="increase"
      label="+"
      onClick={[Function]}
      type="button"
    />
    <Button
      className="decrease"
      label="-"
      onClick={[Function]}
      type="button"
    />
  </div>
</div>
`;
